xbmc/filesystem/File.cpp

```diff
@@ -387,6 +397,21 @@ bool CFile::OpenForWrite(const CURL& file, bool bOverWrite)
   return false;
 }
 
+bool CFile::ReOpen(const CURL& file)
+{
+  if (!m_pFile)
```

Comment:
An alternative could be:
bool CFile::Open(const CURL& file, const bool &bReOpen = false)
....
I think this doesn't break API as well, right?